src/server/json_family.cc

```diff
@@ -1704,18 +1480,13 @@ void JsonFamily::ArrIndex(CmdArgList args, ConnectionContext* cntx) {
   }
 
   auto cb = [&](Transaction* t, EngineShard* shard) {
-    return OpArrIndex(t->GetOpArgs(shard), key, std::move(expression), *search_value, start_index,
-                      end_index);
+    return OpArrIndex(t->GetOpArgs(shard), key, json_path, *search_value, start_index, end_index);
   };
 
   Transaction* trans = cntx->transaction;
-  OpResult<vector<OptLong>> result = trans->ScheduleSingleHopT(std::move(cb));
-
-  if (result) {
-    PrintOptVec(cntx, result);
-  } else {
-    cntx->SendError(result.status());
-  }
+  auto result = trans->ScheduleSingleHopT(std::move(cb));
+  auto* rb = static_cast<RedisReplyBuilder*>(cntx->reply_builder());
+  reply_generic::Send(result, rb);
```

Comment:
be careful not to copy everywhere 🙂 alternatively you can pass by constref or use spans

string family now has a similar overhaul (I use rvalues only because I need to await a future mutably)

https://github.com/dragonflydb/dragonfly/blob/5c7c21b6c5ad5326ff5ef3fdf874d3df7204d1dd/src/server/string_family.cc#L859-L868